### PR TITLE
RAD-149: Split cal_step into L2 & L3

### LINKED ISSUE #363: Split cal_step into L2 & L3
Split the cal_step schema into one for level 2 and one for level 3.

### PR DESCRIPTION
**Pull request by PaulHuwe**
Resolves [RAD-149](https://jira.stsci.edu/browse/RAD-149)

Closes #363 

This PR splits the cal_step schema into L2 and L3 versions.

**Checklist**
- [ ] Schema changes discussed at RAD Review Board meeting
- [x] Added entry in `CHANGES.rst` under the corresponding subsection
- [x] Updated relevant roman_datamodels utilities and tests
- [ ] Passed romancal regression testing on Jenkins / PLWishMaster. Link: https://plwishmaster.stsci.edu:8081/job/RT/job/romancal/693/

### CODE REVIEW

**Comment by PaulHuwe** [comment]
NOTE: these changes require updates to romancal regression test files.

File: src/rad/resources/schemas/l3_cal_step-1.0.0.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    description: |
+      Step in ROMANCAL which resamples each input 2D image based on its WCS and
+      WCS distortion information on a grid that allows the combination of
+      multiple resampled images into a single, undistorted product.
+    type: string
+    enum: ['N/A', 'COMPLETE', 'SKIPPED', 'INCOMPLETE']
+    archive_catalog:
+      datatype: nvarchar(15)
+      destination: [ScienceRefData.s_resample, GuideWindow.s_resample, WFICommon.s_resample]
+propertyOrder: [outlier_detection, skymatch, resample]
```
Comment (ddavis-stsci) [inline src/rad/resources/schemas/l3_cal_step-1.0.0.yaml:39]: Can we also add the flux step to the l3 cal_steps?

Comment (PaulHuwe) [reply src/rad/resources/schemas/l3_cal_step-1.0.0.yaml:39]: Sure, will be sure to merge the flux tickets first then do that. Should it only be in L3, or in both?

Comment (ddavis-stsci) [reply src/rad/resources/schemas/l3_cal_step-1.0.0.yaml:39]: It is the first step in the l3 pipeline, so i think it should only be in the l3 headers.

Comment (schlafly) [reply src/rad/resources/schemas/l3_cal_step-1.0.0.yaml:39]: flux is like outlier detection & sky matching in that it's done on L2 "intermediate" images, so we'd be marking it done on the L2 image models and then propagating that into the L3 image.  i.e., it goes into the L2 and L3 cal_step dictionary.

File: src/rad/resources/schemas/l3_cal_step-1.0.0.yaml
Comment (kdupriestsci) [inline src/rad/resources/schemas/l3_cal_step-1.0.0.yaml:1]: I thought that in Roman all of the r_ and s_ fields went in ScienceRefData and GuideWindow.  Why are the 3 fields in this file going in WFIMosaic?

Also, will these fields appear in any of the RITA files?

**Pull request merged**